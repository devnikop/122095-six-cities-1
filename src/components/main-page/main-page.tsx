import {connect} from 'react-redux';
import * as React from 'react';

import {
  getCurrentCity,
  getSortedOffers
} from '../../reducer/data/selectors';
import withActiveItem from '../../hocs/with-active-item/with-active-item';
import withSortingOptions from '../../hocs/with-sorting-options/with-sorting-options';

import {
  City,
  Offer,
} from '../../types';

import CitiesList from '../cities-list/cities-list';
import Map from '../map/map';
import OfferList from '../offer-list/offer-list';
import SortingOptions from '../sorting-options/sorting-options';

interface Props {
  currentCity: City,
  leaflet,
  sortedOffers: Offer[],
}

const OfferListWrapped = withActiveItem(OfferList);
const SortingOptionsWrapped = withSortingOptions(SortingOptions);

const MainPage:React.FunctionComponent<Props> = (props) => {
  const {
    currentCity,
    leaflet,
    sortedOffers,
  } = props;

  return <>
    <h1 className="visually-hidden">Cities</h1>
    <CitiesList/>
    <div className="cities__places-wrapper">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <h2 className="visually-hidden">Places</h2>
          <b className="places__found">{sortedOffers.length} places to stay in {currentCity}</b>
          {<SortingOptionsWrapped/>}
          <div className="cities__places-list places__list tabs__content">
            {<OfferListWrapped
              offers={sortedOffers}
            />}
          </div>
        </section>
        <div className="cities__right-section">
          <section className="cities__map map">
            <Map
              leaflet={leaflet}
              offers={sortedOffers}
            />
          </section>
        </div>
      </div>
    </div>
  </>;
};

const mapStateToProps = (state, ownProps) =>
  Object.assign({}, ownProps, {
    currentCity: getCurrentCity(state),
    sortedOffers: getSortedOffers(state),
  });

export {MainPage};

export default connect(
    mapStateToProps
)(MainPage);
