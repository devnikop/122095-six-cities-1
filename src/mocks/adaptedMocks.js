const adaptedLoginDataMock = {
  avatarUrl: `/static/avatar/5.jpg`,
  email: `doppervily@yandex.ru`,
  id: 1,
  isPro: false,
  name: `doppervily`,
};

const adaptedOfferMock = {
  bedrooms: 1,
  city: {
    coords: [48.85661, 2.351499],
    zoom: 13,
    name: `Paris`,
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  goods: [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
  host: {
    avatar: `img/avatar-angelina.jpg`,
    id: 25,
    name: `Angelina`,
    isPro: true,
  },
  id: 1,
  images: [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
  isFavorite: true,
  isPremium: false,
  maxAdults: 3,
  place: {
    coords: [48.834610000000005, 2.364499],
    zoom: 16,
  },
  previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
  price: 231,
  rating: 5,
  title: `Canal View Prinsengracht`,
  type: `room`,
};

const adaptedOffersArrayMock = [
  {
    bedrooms: 1,
    city: {
      coords: [48.85661, 2.351499],
      zoom: 13,
      name: `Paris`,
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isPro: true,
    },
    id: 1,
    images: [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
    isFavorite: true,
    isPremium: false,
    maxAdults: 3,
    place: {
      coords: [48.834610000000005, 2.364499],
      zoom: 16,
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    price: 231,
    rating: 5,
    title: `Canal View Prinsengracht`,
    type: `room`,
  },
];

const adaptedReviewsArrayMock = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatarUrl: `img/1.png`,
      id: 4,
      isPro: false,
      name: `Max`,
    },
  },
];

export {
  adaptedLoginDataMock,
  adaptedOfferMock,
  adaptedOffersArrayMock,
  adaptedReviewsArrayMock,
};
