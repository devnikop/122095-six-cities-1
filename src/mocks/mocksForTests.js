const leafletMock = {
  icon() {
    return {
      iconSize: [30, 30],
      iconUrl: `img/pin.svg`,
    };
  },
  map() {
    return {
      setView: jest.fn(),
    };
  },
  tileLayer() {
    return {
      addTo: jest.fn(),
    };
  },
  layerGroup() {
    return {
      addTo: jest.fn(),
    };
  },
  marker() {
    return {
      addTo: jest.fn(),
    };
  },
};

const offersArrayMock = [
  {
    city: `Amsterdam`,
    coords: [52.3909553943508, 4.85309666406198],
    isPremium: true,
    placeName: `Beautiful & luxurious apartment at great location`,
    placeType: `Apartment`,
    price: 120,
    src: `img/apartment-01.jpg`,
  },
  {
    city: `Cologne`,
    coords: [52.369553943508, 4.85309666406198],
    isPremium: false,
    placeName: `Wood and stone place`,
    placeType: `Private room`,
    price: 80,
    src: `img/room.jpg`,
  },
  {
    city: `Brussels`,
    coords: [52.3909553943508, 4.929309666406198],
    isPremium: false,
    placeName: `Canal View Prinsengracht`,
    placeType: `Apartment`,
    price: 132,
    src: `img/apartment-02.jpg`,
  },
  {
    city: `Amsterdam`,
    coords: [52.3809553943508, 4.939309666406198],
    isPremium: true,
    placeName: `Nice, cozy, warm big bed apartment`,
    placeType: `Apartment`,
    price: 180,
    src: `img/apartment-03.jpg`,
  }
];

const citiesMock = [
  {
    coords: [52.370216, 4.895168],
    name: `Amsterdam`,
  },
  {
    coords: [50.850346, 4.351721],
    name: `Brussels`,
  },
  {
    coords: [51.227741, 6.773456],
    name: `Cologne`,
  },
  {
    coords: [50.937531, 6.960279],
    name: `Dusseldorf`,
  },
  {
    coords: [53.551086, 9.993682],
    name: `Hamburg`,
  },
  {
    coords: [48.856613, 2.352222],
    name: `Paris`,
  },
];

const offerMock = {
  isPremium: true,
  placeName: `Beautiful & luxurious apartment at great location`,
  placeType: `Apartment`,
  price: 120,
  src: `img/apartment-01.jpg`,
};

export {
  citiesMock,
  leafletMock,
  offerMock,
  offersArrayMock,
};
