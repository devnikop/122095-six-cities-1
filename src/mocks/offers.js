// const offers = [
//   {
//     city: `Amsterdam`,
//     placeName: `Beautiful & luxurious apartment at great location`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-01.jpg`,
//     price: 120,
//     coords: [52.3909553943508, 4.85309666406198],
//   },
//   {
//     city: `Cologne`,
//     placeName: `Wood and stone place`,
//     placeType: `Private room`,
//     isPremium: false,
//     src: `img/room.jpg`,
//     price: 80,
//     coords: [52.369553943508, 4.85309666406198],
//   },
//   {
//     city: `Dusseldorf`,
//     placeName: `Beautiful & luxurious apartment at great location`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-01.jpg`,
//     price: 120,
//     coords: [52.3909553943508, 4.85309666406198],
//   },
//   {
//     city: `Hamburg`,
//     placeName: `Wood and stone place`,
//     placeType: `Private room`,
//     isPremium: false,
//     src: `img/room.jpg`,
//     price: 80,
//     coords: [52.369553943508, 4.85309666406198],
//   },
//   {
//     city: `Amsterdam`,
//     placeName: `Beautiful & luxurious apartment at great location`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-01.jpg`,
//     price: 120,
//     coords: [52.3909553943508, 4.85309666406198],
//   },
//   {
//     city: `Cologne`,
//     placeName: `Wood and stone place`,
//     placeType: `Private room`,
//     isPremium: false,
//     src: `img/room.jpg`,
//     price: 80,
//     coords: [52.369553943508, 4.85309666406198],
//   },
//   {
//     city: `Brussels`,
//     placeName: `Canal View Prinsengracht`,
//     placeType: `Apartment`,
//     isPremium: false,
//     src: `img/apartment-02.jpg`,
//     price: 132,
//     coords: [52.3909553943508, 4.929309666406198],
//   },
//   {
//     city: `Amsterdam`,
//     placeName: `Nice, cozy, warm big bed apartment`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-03.jpg`,
//     price: 180,
//     coords: [52.3809553943508, 4.939309666406198],
//   }
// ];

// const cities = [
//   {
//     name: `Amsterdam`,
//     coords: [52.370216, 4.895168],
//   },
//   {
//     name: `Brussels`,
//     coords: [50.850346, 4.351721],
//   },
//   {
//     name: `Cologne`,
//     coords: [51.227741, 6.773456],
//   },
//   {
//     name: `Dusseldorf`,
//     coords: [50.937531, 6.960279],
//   },
//   {
//     name: `Hamburg`,
//     coords: [53.551086, 9.993682],
//   },
//   {
//     name: `Paris`,
//     coords: [48.856613, 2.352222],
//   },
//   {
//     name: `Moscow`,
//     coords: [48.856613, 2.352222],
//   },
// ];

// export {
//   offers,
//   cities
// };
