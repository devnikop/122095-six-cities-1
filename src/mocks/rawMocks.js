const rawLoginDataMock = {
  "avatar_url": `/static/avatar/5.jpg`,
  "email": `doppervily@yandex.ru`,
  "id": 1,
  "is_pro": false,
  "name": `doppervily`,
};

const rawOfferMock = {
  "bedrooms": 1,
  "city": {name: `Paris`, location: {latitude: 48.85661, longitude: 2.351499, zoom: 13}},
  "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  "goods": [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
  "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
  "id": 1,
  "images": [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
  "is_favorite": true,
  "is_premium": false,
  "location": {latitude: 48.834610000000005, longitude: 2.364499, zoom: 16},
  "max_adults": 3,
  "preview_image": `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
  "price": 231,
  "rating": 4.8,
  "title": `Canal View Prinsengracht`,
  "type": `room`
};

const rawOffersArrayMock = [
  {
    "bedrooms": 1,
    "city": {name: `Paris`, location: {latitude: 48.85661, longitude: 2.351499, zoom: 13}},
    "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    "goods": [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
    "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
    "id": 1,
    "images": [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
    "is_favorite": true,
    "is_premium": false,
    "location": {latitude: 48.834610000000005, longitude: 2.364499, zoom: 16},
    "max_adults": 3,
    "preview_image": `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    "price": 231,
    "rating": 4.8,
    "title": `Canal View Prinsengracht`,
    "type": `room`
  }
];

const rawReviewsArrayMock = [
  {
    "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "date": `2019-05-08T14:13:56.569Z`,
    "id": 1,
    "rating": 4,
    "user": {
      "avatar_url": `img/1.png`,
      "id": 4,
      "is_pro": false,
      "name": `Max`,
    },
  },
];

export {
  rawLoginDataMock,
  rawOfferMock,
  rawOffersArrayMock,
  rawReviewsArrayMock,
};
