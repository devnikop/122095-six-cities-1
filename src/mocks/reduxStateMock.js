import {
  citiesMock,
  offersArrayMock,
  reviewsArrayMock,
  loginDataMock,
} from './mocksForTests';

import NameSpace from '../reducer/name-spaces';

const StoreMock = {
  data: {
    activeOfferId: 0,
    currentCity: citiesMock[0],
    offers: offersArrayMock,
    sortedOffers: offersArrayMock,
  },
  favorite: {
    favoriteOffers: offersArrayMock,
  },
  review: {
    reviews: reviewsArrayMock,
  },
  user: {
    isAuthorizationRequired: false,
    loginData: loginDataMock,
  },
};

const reduxStateMock = {
  [NameSpace.DATA]: StoreMock.data,
  [NameSpace.FAVORITE]: StoreMock.favorite,
  [NameSpace.REVIEW]: StoreMock.review,
  [NameSpace.USER]: StoreMock.user,
};

export default reduxStateMock;
