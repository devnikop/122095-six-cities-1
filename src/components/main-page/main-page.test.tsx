import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {leafletMock, offersArrayMock} from '../../mocks/mocksForTests';
import {MainPage} from './main-page';

const getFilteredOffers = (city) =>
  offersArrayMock.filter((offer) =>
    offer.city === city);

describe(`Welcome-screen correctly renders`, () => {
  const initialState = {
    cities: [...new Set(offersArrayMock.map((it) => it.city))],
    currentCity: offersArrayMock[0].city.name,
    filteredOffers: getFilteredOffers(offersArrayMock[0].city.name),
  };

  const mockStore = configureStore();
  let store;
  let tree;

  it.skip(`Welcome-screen correctly renders`, () => {
    const leaflet = leafletMock;
    const offers = offersArrayMock;

    store = mockStore(initialState);

    tree = renderer
      .create(<Provider store={store}>
        <MainPage
          currentCity={offers[0].city}
          leaflet={leaflet}
          offers={offers}
        /></Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
