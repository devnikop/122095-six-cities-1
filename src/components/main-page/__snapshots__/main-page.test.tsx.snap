// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome-screen correctly renders 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="cities tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item  "
            href="#"
            onClick={[Function]}
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item  "
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item  "
            href="#"
            onClick={[Function]}
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active "
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities__places-wrapper"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          4
           places to stay in 
          Paris
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
          <select
            className="places__sorting-type"
            id="places-sorting"
            onChange={[Function]}
          >
            <option
              className="places__option"
              value="popular"
            >
              Popular
            </option>
            <option
              className="places__option"
              value="to-high"
            >
              Price: low to high
            </option>
            <option
              className="places__option"
              value="to-low"
            >
              Price: high to low
            </option>
            <option
              className="places__option"
              value="top-rated"
            >
              Top rated first
            </option>
          </select>
        </form>
        <div
          className="cities__places-list places__list tabs__content"
        >
          <article
            className="cities__card place-card"
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="https://es31-server.appspot.com/six-cities/static/hotel/19.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className=" place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    324
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button  button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "72%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  onClick={[Function]}
                >
                  Beautiful & luxurious apartment at great location
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                room
              </p>
            </div>
          </article>
          <article
            className="cities__card place-card"
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="https://es31-server.appspot.com/six-cities/static/hotel/19.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className=" place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    322
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button  button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "72%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  onClick={[Function]}
                >
                  Beautiful & luxurious apartment at great location
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                hotel
              </p>
            </div>
          </article>
          <article
            className="cities__card place-card"
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="https://es31-server.appspot.com/six-cities/static/hotel/19.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className=" place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    80
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button  button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "30%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  onClick={[Function]}
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                private room
              </p>
            </div>
          </article>
          <article
            className="cities__card place-card"
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="https://es31-server.appspot.com/six-cities/static/hotel/19.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className=" place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    80
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button  button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "30%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  onClick={[Function]}
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                apartment
              </p>
            </div>
          </article>
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        >
          <div
            id="map"
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </section>
      </div>
    </div>
  </div>
  }
</main>
`;
